packages/arb-avm-cpp/cavm/ccheckpointstorage.cpp

```diff
@@ -150,13 +152,13 @@ int saveData(CCheckpointStorage* storage_ptr,
     auto value_vector =
         std::vector<unsigned char>(value_str.begin(), value_str.end());
 
-    auto results = transaction->saveValue(key_vector, value_vector);
+    auto results = transaction->saveData(key_vector, value_vector);
 
     if (results.status.ok()) {
         auto status = transaction->commit();
         return status.ok();
     } else {
-        return results.status.ok();
+        false;
     }
```

Comment:
I think the false is missing a `return` 😁. Also I think it'd be nicer to structure this as
```cpp
if (!results.status.ok()) {
    return false;
}
auto status = transaction->commit();
return status.ok();
```
This also applies in the other locations